src/wp-includes/functions.php

```diff
@@ -2074,6 +2075,16 @@ function wp_mkdir_p( $target ) {
 		$dir_perms = 0777;
 	}
 
+	/**
+	 * Fires before the directory is created.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $target    Full path to attempt to create.
+	 * @param int    $dir_perms Directory permissions.
+	 */
+	do_action( 'before_create_directory', $target, $dir_perms );
```

Comment:
Could you add unit test for new action?

Reply:
Sorry I am not familiar with unit tests, there was one at https://core.trac.wordpress.org/attachment/ticket/44083/44083.5.diff but I wasn't sure if I could adopt that one 1/1
Is the old unit test still good, can I add this one?

Reply:
It looks like that older unit test may be doing too much. I can write some unit tests for this and send them through to you @benni1516 🙂

Reply:
Thanks @costdev 🙌